File: src/windows/magma/app/Slice.c
#include "./Slice.h"
class Slice implements Type {
	mut type: Type;
	constructor (mut type: Type) {
		this.type = type;
	}
	mut generate(): &[I8] {
		return "&[" + this.type.generate() + "]";
	}
	mut isFunctional(): Bool {
		return false;
	}
	mut isVar(): Bool {
		return false;
	}
	mut generateBeforeName(): &[I8] {
		return "";
	}
}

File: src/node/magma/app/Slice.ts
import { Type } from "../../magma/api/Type";
class Slice implements Type {
	type: Type;
	constructor (type: Type) {
		this.type = type;
	}
	generate(): string {
		return "&[" + this.type.generate() + "]";
	}
	isFunctional(): boolean {
		return false;
	}
	isVar(): boolean {
		return false;
	}
	generateBeforeName(): string {
		return "";
	}
}
